### PR TITLE
add tbprofiler version 6.2.1

### PR DESCRIPTION
**Pull request by sage-wright**
There's a new version of TB-Profiler! [A few bug fixes were included](https://github.com/jodyphelan/TBProfiler/releases/tag/v6.2.1) (samclip not being used, lineage prediction error, and issue with `update_tbdb`) and I've updated the database used in the docker as well to the most recent commit hash. 

I'm setting this PR as draft for now as the new release is still working its way through the Bioconda PR process.

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

**Comment by erinyoung** [comment]
I don't know if 6.2.1 is going to get on conda.

Do you think it's worthwhile to install the dependencies from sourece?

https://github.com/jodyphelan/TBProfiler?tab=readme-ov-file#manually

**Comment by kapsakcj** [comment]
Jody's working on getting the latest version into bioconda https://github.com/bioconda/bioconda-recipes/pull/47826

**Comment by jodyphelan** [comment]
Just to update you all, I think there is [an issue](https://github.com/bioconda/bioconda-recipes/issues/41025) with bioconda CI which is blocking the uppdate being merged into bioconda.

**Comment by kapsakcj** [comment]
I just resolved README.md conflict, which triggered the CI to test building this docker image.

Should build successfully now that tb-profiler 6.2.1 is available on bioconda 🤞 

@jodyphelan Thanks for the update and for all of your work and maintenance of tb-profiler!

File: tbprofiler/6.2.1/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+- trimmomatic 0.39
+- bwa 0.7.17
+- minimap2 2.16
+- samtools 1.12
+- bcftools 1.12
+- freebayes 1.3.5
+- tqdm 4.32.2
+- parallel v20190522
+- samclip 0.4.0
+- snpeff 5.1
```
[the remark this answers is not in the document]
Comment (sage-wright) [reply tbprofiler/6.2.1/README.md]: Your suggestions have been implemented!

File: tbprofiler/6.2.1/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+LABEL dockerfile.version="1"
+LABEL software="tbprofiler"
+LABEL software.version="${TBPROFILER_VER}"
+LABEL description="The pipeline aligns reads to the H37Rv reference using bowtie2, BWA or minimap2 and then calls variants using bcftools. These variants are then compared to a drug-resistance database."
+LABEL website="https://github.com/jodyphelan/TBProfiler/"
+LABEL license="https://github.com/jodyphelan/TBProfiler/blob/master/LICENSE"
+LABEL maintainer="John Arnn"
+LABEL maintainer.email="jarnn@utah.gov"
+LABEL maintainer2="Curtis Kapsak"
+LABEL maintainer2.email="kapsakcj@gmail.com"
```
Comment (erinyoung) [inline tbprofiler/6.2.1/Dockerfile:24]: Did you want to add yourself as a maintainer?

**Comment by erinyoung** [comment]
It looks like the tests succeeded:

```
#12 [test 3/3] RUN wget -q ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR166/009/ERR1664619/ERR1664619_1.fastq.gz &&     wget -q ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR166/009/ERR1664619/ERR1664619_2.fastq.gz &&     tb-profiler profile -1 ERR1664619_1.fastq.gz -2 ERR1664619_2.fastq.gz -t 2 -p ERR1664619 --txt
#12 10.68 [14:30:21] INFO     Using ref file:                                    db.py:587
#12 10.68                     /opt/conda/share/tbprofiler//tbdb.fasta                     
#12 10.68            INFO     Using gff file:                                    db.py:587
#12 10.68                     /opt/conda/share/tbprofiler//tbdb.gff                       
#12 10.68            INFO     Using bed file:                                    db.py:587
#12 10.68                     /opt/conda/share/tbprofiler//tbdb.bed                       
#12 10.68            INFO     Using json_db file:                                db.py:587
#12 10.68                     /opt/conda/share/tbprofiler//tbdb.dr.json                   
#12 10.80            INFO     Using variables file:                              db.py:587
#12 10.80                     /opt/conda/share/tbprofiler//tbdb.variables.json            
#12 10.81            INFO     Using spoligotype_spacers file:                    db.py:587
#12 10.81                     /opt/conda/share/tbprofiler//tbdb.spoligotype_spac          
#12 10.81                     ers.txt                                                     
#12 10.81            INFO     Using spoligotype_annotations file:                db.py:587
#12 10.81                     /opt/conda/share/tbprofiler//tbdb.spoligotype_list          
#12 10.81                     .csv                                                        
#12 10.81            INFO     Using bedmask file:                                db.py:587
#12 10.81                     /opt/conda/share/tbprofiler//tbdb.mask.bed                  
#12 10.81            INFO     Using barcode file:                                db.py:587
#12 10.81                     /opt/conda/share/tbprofiler//tbdb.barcode.bed               
#12 10.81            INFO     Using rules file:                                  db.py:587
#12 10.81                     /opt/conda/share/tbprofiler//tbdb.rules.txt                 
#12 10.81            INFO     Trimming reads                                   fastq.py:39
#12 13.68 [14:30:24] INFO     Mapping to reference genome                      fastq.py:53
#12 58.84 [14:31:09] INFO     Running variant calling                           bam.py:162
#12 58.87 
Calling variants:   0%|          | 0/73 [00:00<?, ?it/s]
Calling variants:   1%|▏         | 1/73 [00:00<00:31,  2.25it/s]
Calling variants:   4%|▍         | 3/73 [00:00<00:12,  5.51it/s]
Calling variants:   7%|▋         | 5/73 [00:00<00:08,  7.84it/s]
Calling variants:  10%|▉         | 7/73 [00:00<00:06,  9.46it/s]
Calling variants:  12%|█▏        | 9/73 [00:01<00:05, 10.82it/s]
Calling variants:  15%|█▌        | 11/73 [00:01<00:05, 12.03it/s]
Calling variants:  18%|█▊        | 13/73 [00:01<00:05, 10.83it/s]
Calling variants:  21%|██        | 15/73 [00:01<00:05, 11.28it/s]
Calling variants:  23%|██▎       | 17/73 [00:01<00:04, 12.01it/s]
Calling variants:  26%|██▌       | 19/73 [00:01<00:04, 12.35it/s]
Calling variants:  29%|██▉       | 21/73 [00:02<00:03, 13.08it/s]
Calling variants:  32%|███▏      | 23/73 [00:02<00:04, 12.11it/s]
Calling variants:  34%|███▍      | 25/73 [00:02<00:04, 11.75it/s]
Calling variants:  37%|███▋      | 27/73 [00:02<00:03, 12.38it/s]
Calling variants:  40%|███▉      | 29/73 [00:02<00:03, 12.36it/s]
Calling variants:  42%|████▏     | 31/73 [00:02<00:03, 12.12it/s]
Calling variants:  45%|████▌     | 33/73 [00:03<00:03, 11.98it/s]
Calling variants:  48%|████▊     | 35/73 [00:03<00:03, 11.47it/s]
Calling variants:  51%|█████     | 37/73 [00:03<00:03, 11.53it/s]
Calling variants:  53%|█████▎    | 39/73 [00:03<00:02, 12.20it/s]
Calling variants:  56%|█████▌    | 41/73 [00:03<00:02, 12.25it/s]
Calling variants:  59%|█████▉    | 43/73 [00:03<00:02, 12.77it/s]
Calling variants:  62%|██████▏   | 45/73 [00:04<00:02, 12.61it/s]
Calling variants:  64%|██████▍   | 47/73 [00:04<00:02, 12.79it/s]
Calling variants:  67%|██████▋   | 49/73 [00:04<00:01, 13.13it/s]
Calling variants:  70%|██████▉   | 51/73 [00:04<00:01, 13.16it/s]
Calling variants:  73%|███████▎  | 53/73 [00:04<00:01, 12.67it/s]
Calling variants:  75%|███████▌  | 55/73 [00:04<00:01, 12.30it/s]
Calling variants:  78%|███████▊  | 57/73 [00:04<00:01, 12.19it/s]
Calling variants:  81%|████████  | 59/73 [00:05<00:01, 12.45it/s]
Calling variants:  84%|████████▎ | 61/73 [00:05<00:00, 12.91it/s]
Calling variants:  86%|████████▋ | 63/73 [00:05<00:00, 12.75it/s]
Calling variants:  89%|████████▉ | 65/73 [00:05<00:00, 11.53it/s]
Calling variants:  92%|█████████▏| 67/73 [00:05<00:00, 11.18it/s]
Calling variants:  95%|█████████▍| 69/73 [00:05<00:00, 11.78it/s]
Calling variants:  97%|█████████▋| 71/73 [00:06<00:00, 12.15it/s]
Calling variants: 100%|██████████| 73/73 [00:06<00:00, 12.22it/s]
Calling variants: 100%|██████████| 73/73 [00:06<00:00, 11.62it/s]
#12 65.15 
Indexing variants:   0%|          | 0/73 [00:00<?, ?it/s]
Indexing variants:  29%|██▉       | 21/73 [00:00<00:00, 189.22it/s]
Indexing variants:  62%|██████▏   | 45/73 [00:00<00:00, 185.01it/s]
Indexing variants: 100%|██████████| 73/73 [00:00<00:00, 248.80it/s]
#12 65.61 [14:31:16] INFO     Running delly                                      bam.py:88
#12 120.9 [14:32:11] INFO     Running snpEff                                    vcf.py:119
#12 125.8 [14:32:16] INFO     Loading snpEff annotations                        vcf.py:156
#12 126.7 [14:32:17] INFO     Getting genotypes for positions in bed file       bam.py:248
#12 129.0 [14:32:19] INFO     Calculating bamstats                              bam.py:308
#12 131.6 [14:32:22] INFO     Getting qc metrics for regions                    bam.py:342
#12 131.6            INFO     Calculating depth in regions                       bam.py:59
#12 136.4 [14:32:27] INFO     Calculating median depth using samtools           bam.py:178
#12 139.0 [14:32:29] INFO     Getting missing genomic positions                 bam.py:335
#12 139.7 [14:32:30] INFO     Writing outputs                                  output.py:7
#12 139.7            INFO     Writing json file:                              output.py:14
#12 139.7                     /tests/results/ERR1664619.results.json                      
#12 139.7            INFO     Writing text file:                              output.py:18
#12 139.7                     /tests/results/ERR1664619.results.txt                       
#12 139.8            INFO     Profiling finished sucessfully!              tb-profiler:187
#12 DONE 140.0s
```

I'll approve this PR, and we can merge this. @sage-wright , did you want to be listed as a maintainer?

File: tbprofiler/6.2.1/Dockerfile (hunk 2 above)
Comment (sage-wright) [reply tbprofiler/6.2.1/Dockerfile:24]: I did, thank you!

**Comment by sage-wright** [comment]
Just added myself! Thank you @erinyoung!

**Comment by erinyoung** [comment]
Perfect! When the tests finish, I'll merge this PR and deploy to dockerhub and quay

**Comment by erinyoung** [comment]
Thank you for your contribution! You can check the status of your deploy here : https://github.com/StaPH-B/docker-builds/actions/runs/9290468169

**Pull request merged**